src/components/DateRange.js

```diff
@@ -81,14 +102,14 @@ class DateRange extends Component {
     this.props.onRangeFocusChange && this.props.onRangeFocusChange(focusedRange);
   }
   updatePreview(val) {
-    if (!val) {
+    if (!val || !val.wasValid) {
```

Comment:
It should still show the preview because when a user clicks the date, It will select the date. Can you remove the preventing of preview disabling?